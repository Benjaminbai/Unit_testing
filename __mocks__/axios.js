
const mockAxios = jest.requireActual('Axios')

export default mockAxios;