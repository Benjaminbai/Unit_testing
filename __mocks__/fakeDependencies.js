jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))