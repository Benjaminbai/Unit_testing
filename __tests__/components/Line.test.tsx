import '../../__mocks__/fakeDependencies'
import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import Line from '../../components/Charts/Line'

const Props = {
    data: []
};
describe('<Line />', () => {
    test('expect the Line exist', () => {
        const LineComponent = mount(<Router><Line {...Props} /></Router>)
        expect(LineComponent.exists()).toBe(true)
    })

    test('expect Line unmount', () => {
        const LineComponent = mount(<Router><Line {...Props} /></Router>)
        LineComponent.unmount()
        expect(() => {
            LineComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        const LineComponent = render(<Router><Line {...Props} /></Router>)
        expect(LineComponent).toMatchSnapshot()
    })
})