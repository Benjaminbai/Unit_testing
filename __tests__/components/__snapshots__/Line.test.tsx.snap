// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Line /> the source code changed or not 1`] = `
<div>
  <div
    class="echarts-for-react "
    style="height:300px;min-height:340px"
  />
</div>
`;
